hibernate-core/src/main/java/org/hibernate/engine/config/spi/StandardConverters.java

```diff
@@ -7,6 +7,8 @@
 package org.hibernate.engine.config.spi;
 
 
+import org.checkerframework.checker.nullness.qual.PolyNull;
```

Comment:
Remove all the `@PolyNull` annotations in this class and the checks for null `value`, as we now guarantee through the type system that this is non-null

Reply:
Same comment as https://github.com/hibernate/hibernate-orm/pull/6865#discussion_r1247887363